tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -1999,6 +2000,100 @@ public:
                                                     convertVector2RVec);
    }
 
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Return information about the dataframe.
+   /// \return information about the dataframe as string
+   ///
+   /// This convenience function describes the dataframe and combines following information:
+   /// - Number of event loops run, see GetNRuns()
+   /// - Number of total and defined columns, see GetColumnNames() and GetDefinedColumnNames()
+   /// - Name of the columns, see GetColumnNames()
+   /// - Type of the columns, see GetColumnType()
+   /// - Number of processing slots, see GetNSlots()
+   ///
+   /// This is not an action nor a transformation, just a query to the RDataFrame object.
+   ///
+   /// Please note that this is a convenience feature and the layout of the output can be subject
+   /// to change and should not be automatically parsed.
+   ///
+   /// ### Example usage:
+   /// ~~~{.cpp}
+   /// RDataFrame df(10);
+   /// auto df2 = df.Define("x", "1.f").Define("s", "\"myStr\"");
+   /// // Describe the dataframe
+   /// std::cout << df2.Desribe() << std::endl;
+   /// ~~~
+   ///
+   std::string Describe()
+   {
+      // Build set of defined column names to find later in all column names
+      // the defined columns more efficiently
+      const auto columnNames = GetColumnNames();
+      std::set<std::string> definedColumnNamesSet;
+      for (const auto& name : GetDefinedColumnNames())
+          definedColumnNamesSet.insert(name);
+
+      // Get information for the metadata table
+      const std::vector<std::string> metadataProperties = {
+          "Columns in total",
+          "Columns from defines",
+          "Event loops run",
+          "Processing slots"};
+      const std::vector<std::string> metadataValues = {
+          std::to_string(columnNames.size()),
+          std::to_string(definedColumnNamesSet.size()),
+          std::to_string(GetNRuns()),
+          std::to_string(GetNSlots())
+          };
```

Comment:
formatting looks a bit funny, might need a clang-format pass?

Reply:
Good point, haven't run clang-format at all 😬 I'll do this after addressing the other comments.

Reply:
Clang-format does weird stuff :/ /I'll keep it like this now.

Reply:
what do you mean? we try to format everything with clang-format in RDF. for very weird behavior you can enclose blocks of code in `// clang-format off/on` pairs (we do so for doxygen docs, for example).

Reply:
alright, clang-format applied :) i find it not really readable, but i'm happy to following the guidlines ;)